hack/convert-output-to-tsv.py

```diff
@@ -77,29 +71,41 @@ def remove_prefix(text, prefix):
                 if not row:
                     raise RuntimeError(f'Encountered invalid JSON "{line}", aborting.')
 
-                # hles_health_condition: read + copy dog_id and hs_condition and hs_condition_is_congenital
-                # concatenate and write out as pk_name
-                if table_name == "hles_health_condition":
-                    # grab the congenital flag and convert to int
-                    congenital_flag = row.get('hs_condition_is_congenital')
-                    try:
-                        congenital_flag = int(congenital_flag)
-                    except TypeError:
-                        log.info(f"Error, 'hs_condition_is_congenital' is not populated in {table_name}")
-                    row[entity_name] = '-'.join([str(component) for component in [row.get('dog_id'), row.get('hs_condition'), congenital_flag]])
-                elif table_name == "environment":
-                    primary_key_fields = ['dog_id', 'address_1_or_2', 'address_month', 'address_year']
-                    row[entity_name] = '-'.join([str(row.get(field)) for field in primary_key_fields])
+                # generate ids for Firecloud compatibility
+                if args.firecloud:
+                    # hles_health_condition: read + copy dog_id and hs_condition and hs_condition_is_congenital
+                    # concatenate and write out as entity_name, keep original PK
+                    if table_name == "hles_health_condition":
```

Comment:
🔧 I think it's time to refactor this a bit, this is a large conditional with many branches that is difficult to reason about. What if we start by pulling lines 78-92 into a separate method that generates a firecloud formatted row?

Reply:
agreed, there's an enormous amount of if statements nested here. i have some ideas for turning this into either a service object or just a function, but that's likely too high-involvement for now.